topostats/tracing/dnatracing.py

```diff
@@ -493,69 +488,88 @@ def get_splined_traces(self):
                 if not self.splining_success:
                     continue
 
-                spline_average = np.divide(spline_running_total, [step_size, step_size])
+                spline_average = np.divide(spline_running_total, [step_size_px, step_size_px])
                 del spline_running_total
+                spline_average = np.delete(spline_average, -1, 0)
                 self.splined_traces[dna_num] = spline_average
-                # else:
-                #    x = self.fitted_traces[dna_num][:,0]
-                #    y = self.fitted_traces[dna_num][:,1]
-
-                #    try:
-                #        tck, u = interp.splprep([x, y], s=0, per = 2, quiet = 1, k = 3)
-                #        out = interp.splev(np.linspace(0,1,nbr*step_size), tck)
-                #        splined_trace = np.column_stack((out[0], out[1]))
-                #        self.splined_traces[dna_num] = splined_trace
-                #    except ValueError: #if the trace is really messed up just delete it
-                #        self.mol_is_circular.pop(dna_num)
-                #        self.disordered_trace.pop(dna_num)
-                #        self.grains.pop(dna_num)
-                #        self.ordered_traces.pop(dna_num)
 
             else:
```

Comment:
As per above making this handling of linear spines its own method is going to be a better design.